platform/ios/src/UIImage+MGLAdditions.h

```diff
@@ -4,6 +4,6 @@
 
 @interface UIImage (MGLAdditions)
 
-+ (std::unique_ptr<mbgl::SpriteImage>)mbgl_spriteImageFromImage:(UIImage *)image;
++ (std::unique_ptr<mbgl::SpriteImage>)mgl_spriteImage:(UIImage *)image;
```

Comment:
What I meant was that this would become a parameter-less instance method, like this:

```objc
- (std::unique_ptr<mbgl::SpriteImage>)mgl_spriteImage;
```

Within the implementation of an instance method, we have access to all the members of `self`, such as `self.CGImage`. Then you can simply say `annotationImage.image.mgl_spriteImage` to get the `unique_ptr`.